docs/rfcs/20220209_limit_connections_rate.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: Ability to limit connection creation rate per shard
+- Status: in-progress
+- Start Date: 2022-02-09
+- Authors: Vadim Plakhtinskiy
+- Issue: 
+
+# Executive Summary
+
+- Set a limit on the rate with which the shard accepts new connections
+- Set a limit on connection creation rate per IP
+
+
+Similar to KIP-612, adapted to Redpanda's thread per core model
+
+## What is being proposed
+
+New configuration properties (For start we choose global config for connection rate):
+- kafka_max_connection_creation_rate (optional integer, default unset)
+- kafka_max_connection_creation_rate_ovverides (string, default "")
+
+
+When a connection is accepted which exceeds one of the configured
+bounds, it is blocking before reading or writing anything to the socket.
```

Comment:
should it be actually blocking accepting a connection ?

Reply:
Do we know before we accept who the client is?  For a per-client limit we may need to do the accept before we can apply the limits.

Reply:
@jcsp I think not, because I can get ip only from `accept_result` after `socket.accept() `call